woocommerce/woocommerce-android
# Issue/9372 local SKU product search

Closes: #9372 


**"Do Not Merge" label => [FluxC PR needs to be reviewed and merged first](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2789). Make sure to update the FluxC hash in this PR before merging it.**

Note: This improvement can be merged into either 14.7 or 14.8, depending on when we wrap up the review.

Props to @rossanafmenezes - we solved this issue together during a couple pair programming sessions.

### Description


The Product Selection in Order Creation flow currently returns results from the local search before the remote request returns. However, this behavior applies only to "name/description" search and not to the SKU search which always returns a result from the server and does NOT use cache. The goal of this PR is to make this behavior consistent and search in local + remote even for the SKU search option.

### Testing instructions

1. Start the Order Creation flow.
2. Tap on "Add product".
3. Change connection type on your emulator to GPRS.
4. Tap on the search and switch to SKU mode.
5. Type something and notice a result is return immediately (given a match is found in local database).

Test other things and try to break the behavior.

### Images/gif
